tls/s2n_cipher_suites.c

```diff
@@ -1054,10 +1054,34 @@ struct s2n_cipher_suite *s2n_cipher_suite_from_wire(const uint8_t cipher_suite[S
 
 int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_CIPHER_SUITE_LEN])
 {
+    notnull_check(conn);
+
     /* See if the cipher is one we support */
     conn->secure.cipher_suite = s2n_cipher_suite_from_wire(wire);
     S2N_ERROR_IF(conn->secure.cipher_suite == NULL, S2N_ERR_CIPHER_NOT_SUPPORTED);
 
+    /* Verify the cipher was part of the originally offered list */
+    const struct s2n_security_policy *security_policy;
+    GUARD(s2n_connection_get_security_policy(conn, &security_policy));
```

Comment:
You can actually still call `s2n_connection_get_cipher_preferences`! It handles the existence of security policies for you: https://github.com/awslabs/s2n/blob/b1565c67ad3a9038de16395faec301a9f221c019/tls/s2n_connection.h#L320

Reply:
Nice, updated.